File: cli_kanban_python/cli_kanban/model/task.py
"""Task model for Kanban board."""

from dataclasses import dataclass, field
from datetime import datetime
from enum import Enum
from typing import List


class TaskStatus(str, Enum):
    """Task status enumeration."""
    TODO = "todo"
    IN_PROGRESS = "in_progress"
    DONE = "done"

    def next(self) -> "TaskStatus":
        """Get the next status in workflow."""
        if self == TaskStatus.TODO:
            return TaskStatus.IN_PROGRESS
        elif self == TaskStatus.IN_PROGRESS:
            return TaskStatus.DONE
        else:
            return TaskStatus.DONE

    def prev(self) -> "TaskStatus":
        """Get the previous status in workflow."""
        if self == TaskStatus.DONE:
            return TaskStatus.IN_PROGRESS
        elif self == TaskStatus.IN_PROGRESS:
            return TaskStatus.TODO
        else:
            return TaskStatus.TODO


@dataclass
class Task:
    """Represents a Kanban task item."""
    id: int
    title: str
    description: str = ""
    tags: List[str] = field(default_factory=list)
    status: TaskStatus = TaskStatus.TODO
    created_at: datetime = field(default_factory=datetime.now)
    updated_at: datetime = field(default_factory=datetime.now)


@dataclass
class Column:
    """Represents a Kanban column."""
    name: str
    status: TaskStatus
    tasks: List[Task] = field(default_factory=list)


def get_all_columns() -> List[Column]:
    """Get all three columns in order."""
    return [
        Column(name="Todo", status=TaskStatus.TODO),
        Column(name="In Progress", status=TaskStatus.IN_PROGRESS),
        Column(name="Done", status=TaskStatus.DONE),
    ]

File: cli_kanban_go/internal/model/task.go
package model

import "time"

// TaskStatus represents the status column of a task
type TaskStatus string

const (
	StatusTodo       TaskStatus = "todo"
	StatusInProgress TaskStatus = "in_progress"
	StatusDone       TaskStatus = "done"
)

// Task represents a kanban task item
type Task struct {
	ID          int64      `json:"id"`
	Title       string     `json:"title"`
	Description string     `json:"description"`
	Tags        []string   `json:"tags"`
	Status      TaskStatus `json:"status"`
	CreatedAt   time.Time  `json:"created_at"`
	UpdatedAt   time.Time  `json:"updated_at"`
}

// Column represents a kanban column
type Column struct {
	Name   string
	Status TaskStatus
	Tasks  []Task
}

// GetAllColumns returns all three columns in order
func GetAllColumns() []Column {
	return []Column{
		{Name: "Todo", Status: StatusTodo},
		{Name: "In Progress", Status: StatusInProgress},
		{Name: "Done", Status: StatusDone},
	}
}

// NextStatus returns the next status in the workflow
func (s TaskStatus) Next() TaskStatus {
	switch s {
	case StatusTodo:
		return StatusInProgress
	case StatusInProgress:
		return StatusDone
	default:
		return StatusDone
	}
}

// PrevStatus returns the previous status in the workflow
func (s TaskStatus) Prev() TaskStatus {
	switch s {
	case StatusDone:
		return StatusInProgress
	case StatusInProgress:
		return StatusTodo
	default:
		return StatusTodo
	}
}
